src/game.py

```diff
@@ -25,25 +27,25 @@ def get_current_player(self, total_marks_on_board):
         else:
             return self.player_two
 
-    def get_prompt(self, total_marks_on_board):
+    def prompt_for_move(self, total_marks_on_board):
         current_player = self.get_current_player(total_marks_on_board)
-        if self.board.is_full(total_marks_on_board, self.game_board):
-            self.ui.display_message(self.message.game_over_message())
-        else:
-            self.ui.display_message(self.message.prompt_for_move(current_player))
+        self.ui.display_message(self.message.prompt_for_move(current_player))
 
-    def process_user_input(self):
-        position_choice = self.ui.get_user_input(self.message.incorrect_board_input())
-        if self.board.determine_is_spot_taken(self.game_board, position_choice):
-            self.ui.display_message(self.message.spot_taken_message())
-            self.process_user_input()
-        self.board.mark_board(
-            position_choice,
-            self.game_board,
-            self.get_current_player(self.total_marks_on_board),
-        )
-        self.total_marks_on_board = self.board.count_marks(self.game_board)
-        return self.game_board
+    def handle_mark_board(self):
+        move = self.ui.get_user_input()
+        valid_move = self.validator.is_valid_move(self.game_board, move)
+        while not valid_move:
+            self.ui.display_message(self.message.invalid_board_input())
+            move = self.ui.get_user_input()
+            valid_move = self.validator.is_valid_move(self.game_board, move)
+        else:
+            self.game_board = self.board.mark_board(
+                move,
+                self.game_board,
+                self.get_current_player(self.total_marks_on_board),
+            )
+            self.total_marks_on_board = self.board.count_marks(self.game_board)
```

Comment:
is this `else:` doing anything?  This is the first time I've seen a `while - else` pattern.  I actually think its not needed at all.
```suggestion
    def handle_mark_board(self):
        move = self.ui.get_user_input()
        valid_move = self.validator.is_valid_move(self.game_board, move)
        while not valid_move:
            self.ui.display_message(self.message.invalid_board_input())
            move = self.ui.get_user_input()
            valid_move = self.validator.is_valid_move(self.game_board, move)
        self.game_board = self.board.mark_board(move, self.game_board, self.get_current_player(self.total_marks_on_board))
        self.total_marks_on_board = self.board.count_marks(self.game_board)
```